tests/ocp4e2e/helpers.go

```diff
@@ -443,9 +443,68 @@ func (ctx *e2econtext) waitForMachinePoolUpdate(name string) error {
 	return nil
 }
 
-func (ctx *e2econtext) getRemediationsForSuite(s *cmpv1alpha1.ComplianceSuite) int {
+func (ctx *e2econtext) doRescan(s string) {
+	scanList := &cmpv1alpha1.ComplianceScanList{}
+	labelSelector, _ := labels.Parse(cmpv1alpha1.SuiteLabel + "=" + s)
+	opts := &client.ListOptions{
+		LabelSelector: labelSelector,
+	}
+	err := ctx.dynclient.List(goctx.TODO(), scanList, opts)
+	if err != nil {
+		ctx.t.Fatalf("Couldn't get scan list")
+	}
+	if len(scanList.Items) == 0 {
+		ctx.t.Fatal("This suite didn't contain scans")
+	}
+	for _, scan := range scanList.Items {
+		updatedScan := scan.DeepCopy()
+		annotations := updatedScan.GetAnnotations()
+		if annotations == nil {
+			annotations = map[string]string{}
+		}
+		annotations[cmpv1alpha1.ComplianceScanRescanAnnotation] = ""
+		updatedScan.SetAnnotations(annotations)
+
+		bo := backoff.WithMaxRetries(backoff.NewConstantBackOff(apiPollInterval), 180)
+		err := backoff.RetryNotify(func() error {
+			return ctx.dynclient.Update(goctx.TODO(), updatedScan)
+		}, bo, func(err error, d time.Duration) {
+			fmt.Printf("Couldn't rescan after %s: %s\n", d.String(), err)
+		})
+		if err != nil {
+			ctx.t.Fatalf("failed rescan: %s", err)
+		}
+	}
+	var lastErr error
+	err = wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+		suite := &cmpv1alpha1.ComplianceSuite{}
+		key := types.NamespacedName{Name: s, Namespace: ctx.OperatorNamespacedName.Namespace}
+		lastErr = ctx.dynclient.Get(goctx.TODO(), key, suite)
+		if lastErr != nil {
+			return false, nil
+		}
+		if suite.Status.Phase != cmpv1alpha1.PhasePending &&
+			suite.Status.Phase != cmpv1alpha1.PhaseLaunching {
```

Comment:
Hmm this sounds racy, what if the phase goes to running very fast? Wouldn't it be more robust to just check that the state changes after applying the annotation?

Reply:
Should I just check that the phase is not "done" then?

Reply:
I think that would be best. Pending can be extremely fast, Launching should have some delay but...
Checking for != Done seems like safest.